tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/SanitizerDictionary.cs

```diff
@@ -32,52 +36,682 @@ public static ulong GetNextId()
             {
+                #region GeneralRegex
+                // basic RecordedTestSanitizer handles Authorization header
                 new RegisteredSanitizer(
                     new RecordedTestSanitizer(),
-                    "AZSDK001"
+                    "AZSDK0000"
+                ),
+                new RegisteredSanitizer(
+                    new GeneralRegexSanitizer(regex: "SharedAccessKey=(?<key>[^;\\\"]+)", groupForReplace: "key"),
+                    "AZSDK1000"
+                ),
+                new RegisteredSanitizer(
+                    new GeneralRegexSanitizer(regex: "AccountKey=(?<key>[^;\\\"]+)", groupForReplace: "key"),
+                    "AZSDK1001"
+                ),
+                new RegisteredSanitizer(
+                    new GeneralRegexSanitizer(regex: "accesskey=(?<key>[^;\\\"]+)", groupForReplace: "key"),
+                    "AZSDK1002"
+                ),
+                new RegisteredSanitizer(
+                    new GeneralRegexSanitizer(regex: "Accesskey=(?<key>[^;\\\"]+)", groupForReplace: "key"),
+                    "AZSDK1003"
+                ),
+                new RegisteredSanitizer(
+                    new GeneralRegexSanitizer(regex: "Secret=(?<key>[^;\\\"]+)", groupForReplace: "key"),
+                    "AZSDK1004"
+                ),
+                new RegisteredSanitizer(
+                    new GeneralRegexSanitizer(regex: "common/userrealm/(?<realm>[^/\\.]+)", groupForReplace: "realm"),
+                    "AZSDK1005",
+                    "ACS Identity leverages these strings to store identity information."
+                ),
+                new RegisteredSanitizer(
+                    new GeneralRegexSanitizer(regex: "/identities/(?<realm>[^/?]+)", groupForReplace: "realm"),
+                    "AZSDK1006",
+                    "ACS Identity leverages these strings to store identity information."
+                ),
+                new RegisteredSanitizer(
+                    new GeneralRegexSanitizer(regex: "(?:[?&](sig)=)(?<secret>[^&\\\"\\s\\n,\\\\]*)", groupForReplace: "secret"),
+                    "AZSDK1007",
+                    "Common SAS URL Sanitizer. Applies to all headers, URIs, and text bodies."
+                ),
+                #endregion
+                #region HeaderRegex
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("api-key"),
+                    "AZSDK2001"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("x-ms-encryption-key"),
+                    "AZSDK2002"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("Location", value: "https://example.com"),
+                    "AZSDK2003"
+                ),
+                new RegisteredSanitizer(
+                    new HeaderRegexSanitizer("subscription-key"),
+                    "AZSDK2004"
+                ),
```

Comment:
```suggestion
                new RegisteredSanitizer(
                    new HeaderRegexSanitizer("Location", value: "https://example.com"),
                    "AZSDK2003"
                ),
                new RegisteredSanitizer(
                    new HeaderRegexSanitizer("subscription-key"),
                    "AZSDK2004"
                ),
                 new RegisteredSanitizer(
                    new HeaderRegexSanitizer("operation-location", value: "https://example.com"),
                    "AZSDK2004"
                ),
                new RegisteredSanitizer(
                    new HeaderRegexSanitizer("Ocp-Apim-Subscription-Key"),
                    "AZSDK2005"
                ),
```

Reply:
Took this suggestion, but not taking the direct commit because the ID numbers also need to increment properly.